### ISSUE TITLE
default methods on RequestFactory EntityProxies

### ISSUE THREAD

**Issue by lmdg** [gwtproject/gwt#9289]
It would be awesome. 

```java
@ProxyFor(value=User.class, locator=UserLocator.class)
public interface UserProxy extends EntityProxy {
	String getFirstName();
	String getLastName();
	
	default String getFullName() {
		return getFirstName() + " " + getLastName();
	}
}
```

This would depend on #9268 but @SkipInterfaceValidation may also have to be tweaked because it seems that an annotated method is still searched for on the entity by RF server-side.

**Comment by niloc132** [comment]
Initial patch to make these details available to the generator, so that it can optionally skip these methods: https://gwt-review.googlesource.com/#/c/14750 

In terms of a feature, I might suggest that we don't want to support exactly what you have in the issue - getters and setters shouldn't be allowed to be half-implemented by default methods, but instead only support CALL-type methods, formerly supported by categories. Or we split this into two issues - solving CALL methods should be straightforward, but GET/SET will be harder, should we decide to support them in this way.

**Comment by lmdg** [comment]
That's great. Just not sure what you mean by half-implemented. The default methods in this case would be more in the spirit of extension methods (I'm guessing this is what you're referring to with categories a la objc?). Do you mean the actual complexity arises from the "get" and "set"?

**Comment by niloc132** [comment]
By "half-implemented" i mean a getter without a setter or vice versa. Internally, to manage the details about only writing changes back to the server, and freezing beans, etc, autobeans make a distinction between GET/SET type methods, and CALL. Since Autobeans are the basis of requestfactory, it inherits this distinction.

The patch so far only opens the door to recognizing these methods. Next, CALL type methods can be instructed to not require a registered @Category to allow them to work without a wrapped instance. Finally, GET/SET could be considered.

So, at least in my initial attempt at fixing this, GET/SET will be left out. First, AutoBeans and RequestFactory represent a large body of code to audit and check and find various corner cases where this could present a problem. Second, we would need to consider semantics of a default setter but normal getter, or default getter but normal setter, or default getter or setter with no corresponding method. Other cases that might require careful consideration: what happens if someone directly interacts with the underlying Splittable; how to handle an interface has an abstract getter/setter, but a sub-interface declares one or both to be default (just off of the top of my head).

**Comment by lmdg** [comment]
Oh ok, I see, the gritty RF details. Thanks for the work on the fix and the explanation, much appreciated.
